### PR TITLE
Add task for HTTPS API proxy with CSM

### PR DESCRIPTION
**Pull request by fermujica**

### CODE REVIEW

File: modules/ROOT/pages/building-https-proxy.adoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
+= Building an HTTPS API Proxy
+
+API Manager 2.x lets you use the secrets you stored in your secret group to build an HTTPS based API for your CloudHub or Hybrid deployments. +
+By configuring a TLS context for your inbound and outbound traffic, you can ensure your traffic's security.
+
+== Prerequisites
+
+To configure your HTTPS API Proxy using secrets stored in your secret group, you need to meet the following requirements:
+
+* Anypoint Security entitlements. +
```
Comment (chiragnarangmulesoft) [inline modules/ROOT/pages/building-https-proxy.adoc]: I pointed out this before in Slack also. this is not true. Please remove this

File: modules/ROOT/pages/building-https-proxy.adoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+Make sure you have Anypoint Security enabled for your account.
+* Secrets Manager Permissions enabled for your user. +
+If your user is the one creating the TLS context, you need to have the *Write secrets* permission. +
+If your user is the one configuring an existing TLS context for the API Proxy, you need the *Read secrets metadata*, and the *Grant Access to Secrets* permissions. +
```
Comment (chiragnarangmulesoft) [inline modules/ROOT/pages/building-https-proxy.adoc]: @sbehrends @fermujica The only permission in API Manager for configuring a proxy should be Read Secrets metadata, and if there's a different permission for downloading and deploying a proxy then please make it clear.

File: modules/ROOT/pages/building-https-proxy.adoc [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
++
+image::building-https-proxy-7543c.png[]
++
+. Click *Save*
+
+== Deploying the API Proxy
+
+=== CloudHub
+
+After you configured your API Proxy and selected *CloudHub* as the *Proxy deployment target*, your API Proxy will be automatically deployed, with the secrets already configured.
```
Comment (chiragnarangmulesoft) [inline modules/ROOT/pages/building-https-proxy.adoc:59]: @sbehrends @fermujica This needs to mention that secrets are configured as the secure property placeholders and link to Cloudhub  secure property placeholders

File: modules/ROOT/pages/building-https-proxy.adoc [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+=== Hybrid
```
Comment (chiragnarangmulesoft) [inline modules/ROOT/pages/building-https-proxy.adoc:87]: @sbehrends @fermujica This needs to mention that secrets are configured as the secure property placeholders and link to Hybird  secure property placeholders

File: modules/ROOT/pages/building-https-proxy.adoc [hunk 5]
```diff
@@ -0,0 +1,95 @@
+= Building an HTTPS API Proxy
```
Comment (chiragnarangmulesoft) [inline modules/ROOT/pages/building-https-proxy.adoc:1]: @fermujica @sbehrends This doc only applicable for Mule 4 with Cloudhub and Hybrid deployments, and doesn't apply to Mule 3 or RTF. This needs to be explicitly called out and it looks like we removed the old content which was applicable for Mule 3. That still needs to be there

**Review by chiragnarangmulesoft — COMMENTED** [review]
@sbehrends @fermujica I have added a few comments on the doc. Please review and resolved.

File: modules/ROOT/pages/building-https-proxy.adoc (hunk 5 above)
Comment (fermujica) [reply modules/ROOT/pages/building-https-proxy.adoc:1]: Mule 3 is a different topic. 
I think you mean this: https://docs.mulesoft.com/runtime-manager/building-an-https-service

Comment (sbehrends) [reply modules/ROOT/pages/building-https-proxy.adoc:1]: We should add in this document that this feature is supported on Mule 4.x only, and deployment targets supported are CloudHub, Hybrid, and Standalone (not sure if it's required to add Standalone)

File: modules/ROOT/pages/building-https-proxy.adoc (hunk 2 above)
Comment (sbehrends) [reply modules/ROOT/pages/building-https-proxy.adoc]: @fedebrasburg Can you confirm this? I believe that required permission on secret manager is "Grant access to secrets" in order to create a API with TLS Context.

Comment (fedebrasburg) [reply modules/ROOT/pages/building-https-proxy.adoc]: I think `Read secrets metadata` works too

File: modules/ROOT/pages/building-https-proxy.adoc (hunk 1 above)
Comment (fermujica) [reply modules/ROOT/pages/building-https-proxy.adoc]: This was removed.

File: modules/ROOT/pages/building-https-proxy.adoc (hunk 5 above)
Comment (fermujica) [reply modules/ROOT/pages/building-https-proxy.adoc:1]: This was also added: https://github.com/mulesoft/docs-api-manager/pull/21/commits/12b80eae290055132fd70eaa7eda4209bf1cdbf1#diff-c1f77c587f2ddd0f9237d2655dfae320R6

File: modules/ROOT/pages/building-https-proxy.adoc (hunk 4 above)
Comment (fermujica) [reply modules/ROOT/pages/building-https-proxy.adoc:87]: I added this: https://github.com/mulesoft/docs-api-manager/pull/21/commits/8581183ec941191c15a1c7ac284f6884424dbb2f#diff-c1f77c587f2ddd0f9237d2655dfae320R59

File: modules/ROOT/pages/building-https-proxy.adoc (hunk 2 above)
Comment (fermujica) [reply modules/ROOT/pages/building-https-proxy.adoc]: I changed that here: https://github.com/mulesoft/docs-api-manager/pull/21/commits/bfecda31780f413a54a9451169876d11eeb570ff

File: modules/ROOT/pages/building-https-proxy.adoc (hunk 5 above)
Comment (chiragnarangmulesoft) [reply modules/ROOT/pages/building-https-proxy.adoc:1]: This doc is linked from API Manager in Anypoint 
https://docs.mulesoft.com/runtime-manager/building-an-https-service#services-under-api-manager-proxies
We need to add a section to explain for Mule 4 you can use either 
1. Same as Mule 3 
2. The new feature for Mule 4 proxies

Comment (fermujica) [reply modules/ROOT/pages/building-https-proxy.adoc:1]: Sergio mentioned the HTTPS service task for Mule 3 needs an update. I created [DOCS-3331](https://www.mulesoft.org/jira/browse/DOCS-3331) for that.

In the meantime I pushed https://github.com/mulesoft/docs-api-manager/pull/21/commits/de1c77274246e909c14e3a67268036b34f6ede5b to link to those instrcutions

File: modules/ROOT/pages/building-https-proxy.adoc [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+If you want to build and HTTPS Proxy in Mule 3, follow the xref:runtime-fabric::building-an-https-service.adoc[Building an HTTPS Service] tutorial. 
+
+== Prerequisites
+
+To configure your HTTPS API Proxy using secrets stored in your secret group, you need to meet the following requirements:
+
+* Secrets Manager Permissions enabled for your user. +
+To configure an existing TLS context for the API Proxy, you need the *Read secrets metadata* permission.
+Review the xref:anypoint-security::asm-permission-concept.adoc[Secrets Manager Permissions] reference for more information.
+* Make sure your user has an *API Versions Owner* role.
```
Comment (fedebrasburg) [inline modules/ROOT/pages/building-https-proxy.adoc]: `API Versions Owner` isn't available in post crowd world. I would change it for `Make sure your user has at least *Manage API Configuration*`

File: modules/ROOT/pages/building-https-proxy.adoc [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+* A secret group along with a TLS Context in Secrets Manager. +
```
Comment (fedebrasburg) [inline modules/ROOT/pages/building-https-proxy.adoc:17]: Should we add here that the TLS context *must* be of type Mule and the Secret Group *must* be marked as downloadable?

File: modules/ROOT/pages/building-https-proxy.adoc (hunk 6 above)
Comment (fermujica) [reply modules/ROOT/pages/building-https-proxy.adoc]: Updated https://github.com/mulesoft/docs-api-manager/pull/21/commits/604add4830d327f5268bba155e171e8a0ad9a776

File: modules/ROOT/pages/building-https-proxy.adoc (hunk 7 above)
Comment (fermujica) [reply modules/ROOT/pages/building-https-proxy.adoc:17]: Added note: https://github.com/mulesoft/docs-api-manager/pull/21/commits/6fb81cb45fe39874ad98788902722decdd5dddfd#diff-c1f77c587f2ddd0f9237d2655dfae320R19

**Review by sbehrends — APPROVED** [review]

**Pull request merged**